index.js

```diff
@@ -64,11 +64,13 @@ async function fastifyRateLimit (fastify, settings) {
     : defaultMax
 
   // Global time window
-  globalParams.timeWindow = typeof settings.timeWindow === 'string'
-    ? ms.parse(settings.timeWindow)
-    : typeof settings.timeWindow === 'number' && Number.isFinite(settings.timeWindow) && settings.timeWindow >= 0
-      ? Math.trunc(settings.timeWindow)
-      : defaultTimeWindow
+  globalParams.timeWindow = typeof settings.timeWindow === 'function'
+    ? settings.timeWindow
+    : typeof settings.timeWindow === 'string'
+      ? ms.parse(settings.timeWindow)
+      : typeof settings.timeWindow === 'number' && Number.isFinite(settings.timeWindow) && settings.timeWindow >= 0
+        ? Math.trunc(settings.timeWindow)
+        : defaultTimeWindow
```

Comment:
I realize you didn't start this fire, but please put it out:

```suggestion
	const twType = typeof settings.timeWindow
	globalParams.timeWindow = defaultTimeWindow
	if (twType === 'function') {
		globalParams.timeWindow = settings.timeWindow
	} else if (twType === 'string') {
		globalParams.timeWindow = ms.parse(settings.timeWindow)
	} else if (
		twType === 'number' &&
		Number.isFinite(settings.timeWindow) &&
		settings.timeWindow >= 0
	) {
		globalParams.timeWindow = Math.trunc(settings.timeWindow)
	}
```